src/auth/user-record.ts

```diff
@@ -57,19 +58,226 @@ export interface CreateRequest extends UpdateRequest {
   uid?: string;
 }
 
+export interface Metadata {
+  createdAt?: string;
+  lastLoginAt?: string;
+}
+
+export interface MfaInfo {
+  mfaEnrollmentId: string;
+  displayName?: string;
+  phoneInfo?: string;
+  enrolledAt?: string;
+  [key: string]: any;
+}
+
+export interface ProviderUserInfo {
+  rawId: string;
+  displayName?: string;
+  email?: string;
+  photoUrl?: string;
+  phoneNumber?: string;
+  providerId: string;
+  federatedId?: string;
+}
+
+export interface Mfa {
+  mfaInfo?: MfaInfo[];
+}
+
+export interface GetAccountInfoUserResponse extends Metadata, Mfa {
+  localId: string;
+  email?: string;
+  emailVerified?: boolean;
+  phoneNumber?: string;
+  displayName?: string;
+  photoUrl?: string;
+  disabled?: boolean;
+  passwordHash?: string;
+  salt?: string;
+  customAttributes?: string;
+  validSince?: string;
+  tenantId?: string;
+  providerUserInfo?: ProviderUserInfo[];
+  [key: string]: any;
+}
+
+/** Enums for multi-factor identifiers. */
+export enum MultiFactorId {
+  Phone = 'phone',
+}
+
+/**
+ * Abstract class representing a multi-factor info interface.
+ */
+export abstract class MultiFactorInfo {
+  public readonly uid: string;
+  public readonly displayName: string | null;
+  public readonly factorId: MultiFactorId;
+  public readonly enrollmentTime: string;
+
+  /**
+   * Initializes the MultiFactorInfo associated subclass using the server side.
+   * If no MultiFactorInfo is associated with the response, null is returned.
+   *
+   * @param response The server side response.
+   * @constructor
+   */
+  public static initMultiFactorInfo(response: MfaInfo): MultiFactorInfo | null {
+    let multiFactorInfo: MultiFactorInfo | null = null;
+    // Only PhoneMultiFactorInfo currently available.
+    try {
+      multiFactorInfo = new PhoneMultiFactorInfo(response);
```

Comment:
I guess the error occurs when an unexpected response is received. I think we should just expose it to the user, instead of silently ignoring it.

Reply:
We will add new types of second factors in the future. It is possible a developer may be using an older version of the Admin SDK. We shouldn't throw an error in that case.